src/storage/secondary/block/rle_block_iterator.rs

```diff
@@ -0,0 +1,451 @@
+// Copyright 2022 RisingLight Project Authors. Licensed under Apache-2.0.
+
+use bytes::{Buf, Bytes};
+
+use super::{Block, BlockIterator};
+use crate::array::{Array, ArrayBuilder};
+
+pub fn decode_rle_block(data: Block) -> (usize, Block, Block) {
+    let mut buffer = &data[..];
+    let rle_num = buffer.get_u32_le() as usize;
+    let rle_length = std::mem::size_of::<u32>() + std::mem::size_of::<u16>() * rle_num;
+    let rle_data = data[std::mem::size_of::<u32>()..rle_length].to_vec();
+    let block_data = data[rle_length..].to_vec();
+    (rle_num, Bytes::from(rle_data), Bytes::from(block_data))
```

Comment:
We can directly use `data.slice(..)`, therefore to reduce copy. By using `.slice`, we can have two objects referring to a single Bytes memory space, and we don't need extra allocation.

See https://docs.rs/bytes/latest/bytes/ for more information.